compiler/damlc/daml-lf-conversion/src/DA/Daml/LFConversion.hs

```diff
@@ -1016,17 +1032,17 @@ convertExpr env0 e = do
     go env (VarIn GHC_Types "[]") (LType (TypeCon (Is "Char") []) : args)
         = pure $ (, args) $ EBuiltin (BEText T.empty)
     go env (VarIn GHC_Types "[]") args
-        = withTyArg env varT1 args $ \t args -> pure (ENil t, args)
+        = withTyArg env KStar args $ \_env t args -> pure (ENil t, args)
     go env (VarIn GHC_Types ":") args =
-        withTyArg env varT1 args $ \t args ->
-        withTmArg env (varV1, t) args $ \x args ->
-        withTmArg env (varV2, TList t) args $ \y args ->
+        withTyArg env KStar args $ \_env t args ->
+        withTmArg env t args $ \x args ->
+        withTmArg env (TList t) args $ \y args ->
```

Comment:
Arguably the env passed in the continuation should be used, as it has the correct scope. (I realize at the moment it shouldn't matter, but if we change it further down the line.)

```suggestion
        withTyArg env KStar args $ \env1 t args ->
        withTmArg env1 t args $ \x args ->
        withTmArg env1 (TList t) args $ \y args ->
```

Reply:
You are totally right. I was sloppy. Thanks for catching this. I'll go for shadowing the outer `env` though. That reduces the risk of accidentally using the wrong `env`.

Reply:
Ah, that's even better!